### ISSUE TITLE
Allow Listener to have default service type

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-spec#631]
We want it to be possible for a Listener to accept multiple different service types (i.e. a union of service types), and yet still allow a specific service type (i.e. a member of the union) to be inferred from a Listener, when a service declaration does not specify an explicit service type.

**Comment by jclark** [comment]
I think we can solve this by having two attach methods:

* the attach `method` is like now, with a parameter type that specifies all allowed service types (potentially a union); a service declaration with an explicit service type desugars into a call to this
* a new `attachDefault` method has a specific service type (not a union), which is a subtype of the parameter of the `attach` method; a service declaration without an explicit service type desugars into a call to this if the Listener provides it, and otherwise to `attach`

**Comment by sanjiva** [comment]
Interesting approach .. simple solution it seems.

**Comment by jclark** [comment]
@sanjiva @shafreenAnfar Do we still need this given our approach to layering services on top of HTTP?

**Comment by sanjiva** [comment]
Yes we do - we need the HTTP listener to handle both the "RESTService" type and "ServletService" type.

**Comment by jclark** [comment]
But given that we're not using remote methods for POST, why do these need to be different service types?

**Comment by sanjiva** [comment]
After further discussion today - yes we no longer need multiple service types for a single listener type.
